ecs-cli/modules/cli/cluster/cluster_app.go

```diff
@@ -237,8 +239,32 @@ func createCluster(context *cli.Context, awsClients *AWSClients, commandConfig *
 		deleteStack = true
 	}
 
+	tags := make([]*ecs.Tag, 0)
+	if tagVal := context.String(flags.ResourceTagsFlag); tagVal != "" {
+		tags, err = utils.GetTags(tagVal, tags)
+		if err != nil {
+			return err
+		}
+	}
+
+	containerInstanceTaggingSupported := false
+
+	if len(tags) > 0 {
+		// determine if container instance tagging is supported
+		setting, err := getContainerInstanceARNSetting(awsClients.ECSClient)
+		if err != nil {
+			return err
+		}
+		if aws.StringValue(setting.Value) == ecsSettingEnabled {
+			containerInstanceTaggingSupported = true
```

Comment:
Since `containerInstanceTaggingSupported` is being set to effectively the value of the setting when we make the call to ListAccountSettings, could we simply refactor `getContainerInstanceARNSetting()`  to return true or false (plus the usual error handling)? Not sure if the idea was to use that method for other things, but since it's only used to determine whether we can tag instances it might make sense to rename it more descriptively. so something like:
```
ok, err := containerInstanceTaggingSupported(awsClients.ECSClient)
if err != nil {
  return err
}

if ok { ... proceed with tagging etc }
```

Reply:
See: https://github.com/aws/amazon-ecs-cli/pull/731/files#diff-499d2f9fc374c259c658f974cbc87b46R564

Reply:
I have it return the full setting object just so that I can make this log statement: https://github.com/aws/amazon-ecs-cli/pull/731/files#diff-499d2f9fc374c259c658f974cbc87b46R564

Because for ARN settings, its nice if it prints the customer's current IAM identity. 

But I can refactor the function to print this message instead, and then have it just return a bool!

Reply:
As discussed elsewhere, there's also the issue that we only want to make the ListAccountSettings API call if the user specified tags (we don't want to bloat the command with unneeded API calls). And that's the reason there has to be the dangling `
containerInstanceTaggingSupported` bool value.